### PR TITLE
TLS Support

### PR DESCRIPTION
**Pull request by afine**

### CODE REVIEW

File: csd/src/descriptor/service.sdl [hunk 1]
```diff
@@ -204,12 +225,14 @@
...
-        "AUTHENTICATION_TYPE" : "${org.apache.sqoop.security.authentication.type}"
+        "AUTHENTICATION_TYPE" : "${org.apache.sqoop.security.authentication.type}",
+        "TLS_PROTOCOL" : "${org.apache.sqoop.security.tls.protocol}",
+        "SSL_ENABLED" : "${ssl_enabled}"
       }
     },
 
     "topology" : {
-      "minInstances" : "1",
+    "minInstances" : "1",
```
Comment (jarcec) [inline csd/src/descriptor/service.sdl]: This seems as unnecessary change that is just changing indent?

File: csd/src/scripts/rcgen.sh [hunk 2]
```diff
@@ -15,5 +15,9 @@
 
 # for debugging
 set -x
+if [ -n "$SSL_CLIENT_TRUSTSTORE_LOCATION" ]; then
+    export TLS="--tls"
+    echo "set truststore --truststore $SSL_CLIENT_TRUSTSTORE_LOCATION" > sqoop2rc
```
Comment (jarcec) [inline csd/src/scripts/rcgen.sh]: This probably doesn't matter now, but can we perhaps use >> instead of > in case that we will decide to add lines changing the rc file before this line?

File: csd/src/descriptor/service.sdl (hunk 1 above)
Comment (afine) [inline csd/src/descriptor/service.sdl]: will fix

File: csd/src/scripts/rcgen.sh (hunk 2 above)
Comment (afine) [inline csd/src/scripts/rcgen.sh]: will fix to prevent issues in the future
